grpc/profiler_test.go

```diff
@@ -0,0 +1,91 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"google.golang.org/grpc"
+	. "gopkg.in/check.v1"
+
+	"github.com/longhorn/go-common-libs/grpc/profiler/proto"
+	"github.com/longhorn/go-common-libs/test"
+	"github.com/longhorn/go-common-libs/utils"
+)
+
+func (s TestSuite) TestProfilerServiceOperations(c *C) {
+	type testCase struct {
+		op         string
+		portNumber int32
+
+		expect_ret bool
+	}
+
+	testCases := map[string]testCase{
+		"ProfilerOP(...): show": {
+			op:         "show",
+			portNumber: 0,
+			expect_ret: true,
+		},
+		"ProfilerOP(...): enable": {
+			op:         "enable",
+			portNumber: 55555,
+			expect_ret: true,
+		},
+		"ProfilerOP(...): disable": {
+			op:         "disable",
+			portNumber: 0,
+			expect_ret: true,
+		},
+	}
+
+	grpcServerPort, err := utils.GenerateRandomNumber(50000, 55000)
+	if err != nil {
+		c.Fatalf("Failed to generate random number: %v", err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	client, closeFunc := server(ctx, grpcServerPort)
```

Comment:
If you are using a callback to shutdown the server, then you do not need to use ctx and vice versa. I would suggest using ctx because you can tear down every asset used in the test case after the test case is finished.

Reply:
OK, I keep the ctx and remove the close callback. Thanks for the explanation.